native-rating/implementation.md

```diff
@@ -0,0 +1,212 @@
+of the carrier definition in the Carrier API app **exactly**. In a newly created Carrier API app, this would be in the file
+located at `./src/definitions/demo-carrier/index.ts`. It is possible that your carrier definition does not have an ApiCode,
+and if it doesn't, you'll need to add it at the root of the carrier:
+
+```typescript
+export const DemoCarrier: Carrier = {
+  // DO NOT CHANGE THIS ID AFTER PUBLISHING
+  Id: "23a9753b-159b-4dd0-ac0e-6be065dccdd0",
+  Name: "Demo Carrier",
+  ...
+  ApiCode: "demo_carrier",
+  ...
+};
+```
+
+The next step is to add a `NativeRating` property to the same carrier definition, specifying the Native Rating appId and
+the id of the default rate card that should be used by requests from the platform.
+
+```typescript
+export const DemoCarrier: Carrier = {
+  // DO NOT CHANGE THIS ID AFTER PUBLISHING
+  Id: "23a9753b-159b-4dd0-ac0e-6be065dccdd0",
+  Name: "Demo Carrier",
+  ...
+  NativeRating: {
+    AppId: "c1d1cf8d-71b7-4939-a04d-2f4e46904231",
+    DefaultRateCard: "default-rate-card"
+  }
+};
+```
+
+The `AppId` can be found in the `package.json` file of the Native Rating app. The default rate card can either be the id
+of the Connect managed rate card specified in the `./src/demo-data.ts` file or a different rate card managed manually
+via the Native Rating api. The default rate card can be changed at any time so even if you're using a manually managed
+rate card, it may be worth starting out with the Connect managed rate card to ensure everything is working correctly
+in ShipStation or ShipEngine and changing it after that.
+
+When those changes have been made, your Carrier API app changes are ready to be published but you'll need to publish
+the Native Rating app first.
+
+### Publishing the Native Rating app
+
+When the implementation is complete, you can use the Connect CLI to publish the Native Rating app to the development
+environment for live testing. To do this, ensure that the package is built and bundled by running the following:
+
+```bash
+npm run build
+npm run bundle
+```
+
+When that is finished, publish the app using the Connect CLI:
+
+```bash
+connect publish
+```
+
+This will ask for your api key which you should have received before building your integration. If you do not have one,
+contact the [ShipEngine Connect team](mailto:connect@shipengine.com). The publish process can take some time, but once
+it is complete your rates will be available directly via the Native Rating service.
+
+If you are linking the Native Rating app to a Carrier API app, you can publish those changes now.
+
+:::warning Note
+Please note that publishing the Native Rating app from the Connect CLI makes the carrier available in Native Rating Service that can be accessed from development environments, like DDEs. To get the carrier rates in production, the carrier data needs to be [published](https://connect-rating.shipengine.com/app/publishing) via Native Rating.
```

Comment:
This link goes to the NR UI, which seems out of place because I don't think the audience of these docs will be expected to log in at the moment.

Reply:
removed UI links.